architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,224 @@
+
+##### Access to cloud credentials and cloud
+
+Applications RP requires AWS and Azure credentials for accessing and managing resources in cloud. It fetches credentials using UCP Secret Provider library. 
+Credentials are not available for retrieval through API. Also, RP supports federated identities which enables robust security practices. 
+
+Using these credentials, Applications RP can create other resources in AWS and Azure. The RP also can create managed identities for azure which will decide who can deploy and run user code.  
+
+#### Access to cluster
+
+The RP can create kubernetes resources and manage them on behalf of the user. It can, for example, create a container based on the image provided by the user, which can in turn execute arbitrary code, and create other resources in the cluster where Radius is running, or on a different cluster. 
+
+#### Exposing User Application to Internet
+
+The Radius RP can create ingress Kubernetes objects, which can expose a kubernetes service to internet. 
+
+#### Bicep Recipe execution
+
+`Recipes` are arbitrary Bicep or Terraform code which can provision infrastructure resources. In order to exceute a Bicep recipe, Applications RP's Recipe Engine first downloads the recipe from a OCI compliant registry over HTTP. Then it requests UCP to deploy the bicep. 
+
+#### Terraform Recipe execution
+
+Terraform recipes are download from internet too, since they are available as public modules. These recipes are downloaded onto an empty directory `/terraform` which is mounted into the applications RP pod. Then the installed terraform executable executes these recipes in the current directory. Terraform communicates with AWS and Azure as needed to deploy resources.
+
+#### Data Serialization / Formats
+
+We use custom parsers to parse Radius-related resource IDs and do not use any other custom parsers. The custom parser that parses Radius resource IDs has its own security mechanisms that don't accept anything other than a Radius resource ID. 
+
+We also use json for data interchange between Applications RP and other services. We use golang standard libraries to parse json. 
+
+## Clients
+
+**UCP** UCP is the client of Applications RP.  It forwards all requests related to Applications and several of Application resources deployment to Applications RP.
+
+## Trust Boundaries
+
+We have a few different trust boundaries for the Controller component:
+
+- **Kubernetes Cluster**: The overall environment where the Applications RP  operates and receives requests from the UCP.
+- **Namespaces within the Cluster**: Logical partitions within the cluster to separate and isolate resources and workloads.
+
+The Applications RP component lives inside the `radius-system` namespace in the Kubernetes cluster where it is installed. UCPD also resides within the same namespace.
+
+The Kubernetes API Server, with which Applications RP interacts, runs in the `kube-system` namespace within the cluster.
+
```

Comment:
What trust model do we apply to requests coming in to Applications RP?

- We assume that the actor is an authorized user, and has authorization make the request. 

How we do we enforce this?

- Action Item: We should enforce that traffic is only accepted from UCP.
- UCP Action Item: We should enforce authentication and authorization in UCP.